docs/table-upgrade.md

```diff
@@ -0,0 +1,101 @@
+# Table Upgrade logic and data structures
+The Hive Metastore migration process will upgrade the following Assets:
+- Tables ond DBFS root
+- External Tables
+- Views
+
+We don't expect this process to be a "one and done" process. This typically is an iterative process and may require a few runs.
+
+We suggest to keep track of the migration and provide the user a continuous feedback of the progress and status of the upgrade.
+
+The migration process will be set as a job that can be invoked multiple times.
+Each time it will upgrade tables it can and report the ones it can't.
+
+## Common considerations
+1. One view per workspace summarizing all the table inventory and various counters
+1. By default we create a single catalog per HMS (<prefix (optional)>_<workspace_name>), happens at the account level.
+1. Workspace Name would be set up as part of the installation at the account level.
+1. Consider other mappings of environments/database to catalog/database.
+1. We have to annotate the status of assets that were migrated.
+1. We will roll up the migration status to the workspace/account level. Showing migration state. 
+1. Aggregation of migration failures
+   1. View of object migration:
+
+   | Object Type | Object ID | Migrated | Migration Failures |
+   |----|----|----|----|
+   |View|hive_metastore.finance.transaction_vw|1|[]|
+   |Table|hive_metastore.finance.transactions|0|["Table uses SERDE: csv"]|
+   |Table|hive_metastore.finance.accounts|0|[]|
+   |Cluster|klasd-kladef-01265|0|["Uses Passthru authentication"]|
+
+1. By default the target is the target_catalog/database_name
+1. We need to set a repository for upgrade mapping. It will be used to override the default database mapping.
+
+   | Source Database | Target Catalog | Target Database |
+   |----|----|----|
+   |finance| de_dev | finance |
+   |hr | de_dev | human_resources|
+   |sales | ucx-dev_ws | sales |
+1. We need to provide a UI or other option to override these values.
+1. By default we copy the table content (CTAS)
+1. Allow skipping individual tables/databases
+1. Explore sizing tables or another threshold (recursively count bytes)
+1. By default we copy the table into a managed table/managed location
+1. Allow overriding target to an external table
+1. We should migrate ACLs for the tables (where applicable). We should highlight cases where we can't (no direct translation/conflicts)
+1. We should consider automating ACLs based on Instance Profiles / Service Principals and other legacy security mechanisms
+
+## Tables (Parquet/Delta) on DBFS root
```

Comment:
Can we keep Managed Tables separate from External Tables.  This is how it maps to our upgrade steps.

External tables are easier and lower risk.  We should be able to deliver on those faster.